service-api/app/src/App/src/Service/Lpa/RemoveLpa.php

```diff
@@ -7,41 +7,49 @@
 use App\DataAccess\Repository\UserLpaActorMapInterface;
 use App\DataAccess\Repository\ViewerCodesInterface;
 use App\Exception\NotFoundException;
+use DateTime;
+use Exception;
 use Psr\Log\LoggerInterface;
 
-class DeleteLpa
+class RemoveLpa
 {
     private LoggerInterface $logger;
     private UserLpaActorMapInterface $userLpaActorMapRepository;
     private ViewerCodesInterface $viewerCodesRepository;
+    private LpaService $lpaService;
 
     public function __construct(
         LoggerInterface $logger,
         UserLpaActorMapInterface $userLpaActorMapRepository,
-        ViewerCodesInterface $viewerCodesRepository
+        ViewerCodesInterface $viewerCodesRepository,
+        LpaService $lpaService
     ) {
         $this->logger = $logger;
         $this->userLpaActorMapRepository = $userLpaActorMapRepository;
         $this->viewerCodesRepository = $viewerCodesRepository;
+        $this->lpaService = $lpaService;
     }
 
     /**
-     * Deletes an LPA from a users account
+     * Removes actor association from all viewer codes created by a user
+     * and removes an LPA from a users account
      *
      * @param string $userId The user account ID that must correlate to the $token
-     * @param string $token UserLpaActorToken that map an LPA to a user account
-     * @return ?array A structure that contains processed LPA data and metadata
+     * @param string $token  UserLpaActorToken that map an LPA to a user account
+     *
+     * @return array A structure that contains processed LPA data and metadata
+     * @throws Exception
```

Comment:
```suggestion
     * @throws NotFoundException
```
Or maybe it can throw others. It's probably good to be explicit